pkg/traces/exporter/exporter.go

```diff
@@ -262,10 +255,46 @@ func (t *TraceExporter) setNewGlobalProvider() {
 	osquerygotraces.SetTracerProvider(newProvider)
 
 	if t.provider != nil {
+		// shutdown still gets called even though the span processor is unregistered
+		// leaving this in because it just feel correct
+		t.provider.UnregisterSpanProcessor(t.bufSpanProcessor)
 		t.provider.Shutdown(t.ctx)
 	}
 
 	t.provider = newProvider
+
+	// now that we have set up new provider, see if we need a new exporter and therefore
+	// a new child processor for the buf span processor
+
+	if t.bufSpanProcessor.HasProcessor() && t.ingestUrl == t.knapsack.TraceIngestServerURL() {
+		// already have processor pointing at the same url, so no need to recreate
```

Comment:
I think we'd also want to recreate if `t.disableIngestTLS` or `t.batchTimeout` changed?

Reply:
yep, you're correct ... been trying to find a good way to do this, now I'm thinking of adding another func for setNewBatchProvider or something along those lines, I'll let you know when I update

Reply:
Maybe instead of just `needsNewProvider` bool in `FlagsChanged`, you could have `needsNewProvider` and `needsNewExporter` and pass both of those in to this function? And then you could still set `t.ingestUrl` inside of `FlagsChanged` instead of having to remember to do it here

Reply:
added a bool param to set new provider for rebuilding the exporter, the only time we explicitly don't to rebuild the exporter is when we set the osquery attributes, since it's async and could whomp the buffered spans if exporter is rebuilt